libraries/AP_Generator/AP_Generator_IE_FuelCell.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include "AP_Generator_Backend.h"
+
+#ifndef GENERATOR_ENABLED
+#define GENERATOR_ENABLED !HAL_MINIMIZE_FEATURES
+#endif
+
+#if GENERATOR_ENABLED
+
+#include <AP_Param/AP_Param.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <GCS_MAVLink/GCS.h>
+
+#define TERM_BUFFER            12  // Max length of term we expect
+#define HEALTHY_TIMEOUT_MS   5000  // Time for driver to be marked un-healthy
+
+class AP_Generator_IE_FuelCell : public AP_Generator_Backend
+{
+
+public:
+    // Constructor
+    AP_Generator_IE_FuelCell(AP_Generator &frontend, AP_Generator::GenType type);
+
+    // Initialize the fuel cell driver
+    void init(void) override;
+
+    // Check if readings are healthy
+    bool healthy(void) const override { return _state.healthy; }
+
+    // Check for arming
+    bool pre_arming_checks(char *failmsg, uint8_t failmsg_len) override;
+
+    // Check for failsafes comparing to action bitmasks set by operator
+    AP_BattMonitor::BatteryFailsafe update_failsafes(void) const override;
+
+
+protected:
+
+    // Fuel cell model type
+    AP_Generator::GenType _type;
+
+    // Pointer to serial uart
+    AP_HAL::UARTDriver *_uart = nullptr;
+
+    // IE fuel cell running states and corresponding lookup table
+    enum class FUEL_CELL_STATE {
```

Comment:
Given this is enum-class within `AP_Generator_IE_FuelCell`, this can just be `State`.